### ISSUE TITLE
New Terminal Recording / gif for README

### ISSUE THREAD

**Issue by V1rgul** [bhilburn/powerlevel9k#370]
As we're looking for a Logo, I thought it might be great to update the screenshots in the Readme.

Some of them are quite old and/or not using a patched-font, which is a plus of p9k.
So if somebody changes them, he'll hopefully make them hdpi-ready :)

Ideally, they should be 1776px width for 2x scaling, as Github's Readme width is 888px.

**Comment by bhilburn** [comment]
Great idea, @V1rgul!

Does anyone have experience recording terminals that can offer some guidance on how to go about it?

**Comment by polarathene** [comment]
I've done a GIF recording on OSX before for a [PR on iTerm2](https://github.com/gnachman/iTerm2/pull/277) with a feature I implemented. I don't have a HiDPI screen on me however(nor a Mac, was at previous company), not sure if I can do such a recording.

There is [ASCIInema](https://asciinema.org/), I haven't tried this yet but seen it around to show off terminal animations. Anything that can record the screen should be fine, you can convert video's into animated gifs easily([Giphy](http://giphy.com/) is pretty neat for that I think).

**Comment by V1rgul** [comment]
I tried redoing the first capture, but I have some annoying glitches with a font that big.
How does this image render on your screen at 50% zoom (in a new tab) ?
![image](https://i.imgur.com/7vXUaFo.png)

**Comment by polarathene** [comment]
@V1rgul looks fine to me on 1080p display at 100% or 50%. Preview as well.

**Comment by bhilburn** [comment]
Preview & full image look fine for me (other than the drawing glitch on the right side in the time segment).

**Comment by bhilburn** [comment]
Looks like a combination of asciinema and this tool will do the trick: https://github.com/tav/asciinema2gif

I might give this a whirl soon.

Thoughts on what segments should be shown off in the gif, anyone (other than the standard `dir`, `vcs`, `status`, `time`)?

@V1rgul @polarathene

**Comment by polarathene** [comment]
The current example is quite nice?

![current example](https://camo.githubusercontent.com/80ec23fda88d2f445906a3502690f22827336736/687474703a2f2f692e696d6775722e636f6d2f777942565a51792e676966)

Not sure how much effort would be involved but something like [oh my git](https://github.com/arialdomartini/oh-my-git) might be popular? 

I'm not sure what other users are doing but I tend to prefer not too many segments, contextual ones are great, but good theme/colours also makes for nice eye candy, which for some themes I've completely dismissed their functionality when browsing through so many :(

**Comment by bhilburn** [comment]
Okay, so I tried to do this with asciinema, and unfortunately the font isn't rendering correctly. It's the classic "gap between the segments" issue that we see with improperly configured terminals: 
https://asciinema.org/a/6jebo7t3zff3hpwle1623kn11

It looks like asciinema *does* support powerline, though, based on these former bugs:
https://github.com/asciinema/asciinema.org/issues/116
https://github.com/asciinema/asciinema.org/issues/161

Also, you can see in the video that it is even rendering the awesome font glyphs correctly... it's just those segment separators that aren't getting drawn right. Any help, @sickill, by chance? =)

**Comment by polarathene** [comment]
Sometimes that issue can be corrected with a different font or font size. I had a similar issue if I recall with iTerm2 and Hack font, I patched the font myself with macOS app Glyphs(free trial).

Alternatively if someone has a HiDPI display they could perhaps do a screen recording, crop it and convert to gif?

**Comment by sickill** [comment]
@bhilburn hey! Can you define "gap between segments"? Where is it occurring on the screen in this asciinema recording?

**Comment by sickill** [comment]
@bhilburn do you refer to this square that's displayed in several places?

**Comment by bhilburn** [comment]
@sickill - Actually, I'm a little confused at the moment. I really thought when I posted that comment that the recording properly showed the FontAwesome icons, but now they appear to be the squares instead. Did something change just in the last few days?

The original problem, which still exists, is with the triangular segment separator. In the videos, it's either not the right height or there is a "gap" between that character in the one next to it. Here are two images, one at normal zoom and one zoomed in:

![gap](https://cloud.githubusercontent.com/assets/569571/22348487/d1c78d74-e3d9-11e6-9862-3eeefebed028.PNG)

![zoomed](https://cloud.githubusercontent.com/assets/569571/22348490/d3544448-e3d9-11e6-866b-f2ba539794e8.PNG)

**Comment by polarathene** [comment]
That issue is the one I fixed with my patch to the `Hack` font I use. But it only works for the font size I adjusted it for from memory. This can vary based on terminal used and a few other causes too from memory, so not really fixable for everyone :/

**Comment by sickill** [comment]
@bhilburn well, asciinema never had support for FontAwesome (it supports Poweline-Symbols font though).

As @polarathene wrote, this is really tricky stuff. If you develop desktop terminal emulator it's probably easier to make the font sizes, paddings, alignments right. But on the web, we support all major browsers and many OSes, and they all render fonts in slightly different ways. I'm afraid we can't do much at the moment (rendering to <canvas> could improve the situation because we would have more precise control on the positions of rendered characters).

**Comment by bhilburn** [comment]
@sickill - Ah, thanks for providing some input! I really appreciate it!

@polarathene - Makes sense. Based on your & @sickill's response, it sounds like we'll need to find another option.

@V1rgul - What did you use to make your recording? I can give it a go and see if I see the same issues with large fonts.

**Comment by bhilburn** [comment]
Okay, I proceeded with the release without this update, but now I want to circle back around to this.

So, really what we need is effectively a way to record a screencast and create a gif from it. The original recording was done by @natemccurdy. 

What did you use, @natemccurdy, because that seemed to work well?

**Comment by natemccurdy** [comment]
@bhilburn I used Recordit

http://recordit.co/

**Comment by dritter** [comment]
I made my screencasts (in #344) with [ScreenFlow](https://www.telestream.net/screenflow/overview.htm) for Mac. Worked pretty well.

**Comment by V1rgul** [comment]
Here's mine (taken with http://www.screentogif.com)
![](https://gifyu.com/images/p9k_.gif)

**Comment by V1rgul** [comment]
realised it's too small without using http://userstyles.org/styles/108591/github-wide so remade it
![](https://gifyu.com/images/p9k_2_.gif)

**Comment by bhilburn** [comment]
So I really like @V1rgul's most recent gif, _a lot_, with the exception of the poor rendering of the segment transitions in the right prompt - they aren't as visible on the left. Some users might see that and think it's an issue with P9k rather than realizing it's a limitation of the terminal recorder.

Any thoughts on how to remove that effect? Is there a way to specify a font to render with, by chance? :-/

**Comment by V1rgul** [comment]
@bhilburn It's a screen recording so you can't change the font afterwards. There's always rendering problems on the right-to-left arrows with any of the monospaced fonts I have :/

**Comment by V1rgul** [comment]
Ok, so changing my font to a Powerline patched (and adjusted) font found here: https://github.com/daagar/powerline-fonts solved the misaligned left arrow problem.

In the `Dejavu Sans Mono for Powerline Book` there's already a symbol at `0xf321` which is supposed to be `VCS_COMMIT_ICON`.
What I did was edit the awesome-terminal-fonts here: https://github.com/gabrielelana/awesome-terminal-fonts/blob/master/build.sh#L13 to relocate octicons further.
I changed
```sh
./scripts/fu-relocate ./fonts/octicons-regular.ttf --save-as='.work/octicons-regular-relocated.ttf' --to='0xf300' 2> /dev/null
```
to
```sh
./scripts/fu-relocate ./fonts/octicons-regular.ttf --save-as='.work/octicons-regular-relocated.ttf' --to='0xf330' 2> /dev/null
```
Now my arrows are aligned correctly and I can still use the fontconfig icons ;)

![gif](https://gifyu.com/images/p9knew.gif)
